### PR TITLE
#6035 - Extend Markdown Support Pt. 1

### PR DESCRIPTION
**Pull request by Miaplacidus**
The work in this PR adds styles for and confirms functionality for features in this list:

- [x] Headers (all 6, with proper spacing below) `# ## ###...`
  - Headers 1-5 should reflect our design system
  - Headers 6 should simply maintain the styling for H5 as we don't have a detailed H6. 
- [x] Horizontal Rule / Divider line (with proper spacing above + below)
- Inline Typography: 
  - [x] Bold
  - [x] Italic
  - [x] Bold + Italic (together)
  - [x] Strike-through
  - [x] underline
- [x] Block Quotes (and nest-able) `>` `> >`
- Lists
  - [x] Ordered
  - [x] Unordered
  - [x] Lettered (consitutional style)
  - [ ] Todo lists
- Code
  - [x] Inline (single tick)
  - [x] Block (wrapping triple tick)
- [x] Tables
  - [x] Inline Typography properly applying to text in table cells
- Links
  - [x] Standard MD format for hyperlinks (with text)
  - [x] Auto-convert pasted URL into clickable link
- Images
  - [x] D&D into editor, upload and insert MD, allow update of alt-text
  - [x] Standard MD format, with alt text (image ref to external server)
  - [x] Image Link (clickable image)
- [x] Special Character Escaping (https://www.markdownguide.org/basic-syntax/#characters-you-can-escape)

The functionality of checkboxes works, but the styling will be left to a different card. Please ignore that for this review. 

## Link to Issue
Closes: #6035 

## Description of Changes
- see above list for implemented styling.

## Test Plan
- Input markdown for the features in the list above and check that they look/work as expected

## Other Considerations
- currently discussing work around checkboxes

### CODE REVIEW

**Review by Israellund — APPROVED** [review]
lgtm

File: packages/commonwealth/client/styles/components/react_quill/markdown_formatted_text.scss [hunk 1]
```diff
@@ -12,78 +13,198 @@
...
+    @include b2;
+    color: $primary-500;
+  }
+
+  h1 {
+    font-family: $font-family-silka;
+    font-weight: 600;
+    font-size: 24px;
+    line-height: 36px;
+    letter-spacing: -0.24px;
```
Comment (masvelio) [inline packages/commonwealth/client/styles/components/react_quill/markdown_formatted_text.scss:37]: image

Sizing of headers does not seem to be accurate. Per ticket `Headers 1-5 should reflect our design system`. 

Figma with headers: https://www.figma.com/file/yCUKk46hg9Fy3EXGkSoaKT/%F0%9F%9A%A7-Foundation?type=design&node-id=31-178&mode=design&t=htdDyuWwikW9yR7X-0

**Review by masvelio — APPROVED** [review]
Only one question about headers size. Other than that lgtm

File: packages/commonwealth/client/styles/components/react_quill/markdown_formatted_text.scss (hunk 1 above)
Comment (jnaviask) [reply packages/commonwealth/client/styles/components/react_quill/markdown_formatted_text.scss:37]: cc @Miaplacidus then this PR is good to go

Comment (Miaplacidus) [reply packages/commonwealth/client/styles/components/react_quill/markdown_formatted_text.scss:37]: @jnaviask @masvelio 
The description is a bit confusing. Per a conversation between myself, @zakhap, and @jessmart1213, the header sizing is per the Figma doc attached to the ticket. In addition, headers 3 - 6 all have the same styling.

**Pull request merged**